### PR TITLE
fix: test_sync_from_achival_node

### LINKED ISSUE #2366: test_sync_from_achival_node fails
```    
GC Error: invalid refcount for `EsghDH6x5odS9mPfPi29ziN1AExsjdrjW6nZJCDApaQX`, expected 1, found 2 }
thread 'test_sync_from_achival_node' panicked at 'assertion failed: false', chain/client/src/client.rs:787:21
```
Seems to be related to #2084

### PR DESCRIPTION
**Pull request by bowenwang1996**
When we clear block data in gc the way to tell whether there is a fork is wrong. We should look at block refcount instead of whether there are multiple blocks at a height, since there can be a fork with heads on different heights. However, this doesn't fix the deeper issue #2386. Resolves #2366.

Test plan
---------
`test_sync_from_achival_node` passes

### CODE REVIEW

File: chain/chain/src/chain.rs [hunk 1]
```diff
@@ -628,18 +628,10 @@ impl Chain {
...
                 let prev_hash = chain_store_update.get_block_header(block_hash)?.prev_hash;
-                if *chain_store_update.get_block_refcount(&prev_hash)? != 1 {
-                    return Err(ErrorKind::GCError(format!(
-                        "invalid refcount for {:?}, expected 1, found {:?}",
-                        block_hash,
-                        chain_store_update.get_block_refcount(&prev_hash).unwrap()
-                    ))
-                    .into());
+                // break when there is a fork
+                if *chain_store_update.get_block_refcount(&prev_hash)? > 1 {
```
Comment (Kouprin) [inline chain/chain/src/chain.rs:633]: return Err if block refcount == 0

File: chain/chain/src/chain.rs [hunk 2]
```diff
@@ -628,18 +628,10 @@ impl Chain {
...
+                    break;
                 }
                 chain_store_update.clear_block_data(trie.clone(), *block_hash, false)?;
```
Comment (Kouprin) [inline chain/chain/src/chain.rs:636]: debug assert that blocks_current_height.len() == 1

File: chain/client/tests/bug_repros.rs [hunk 3]
```diff
@@ -159,7 +157,7 @@ fn test_sync_from_achival_node() {
             false,
             false,
             epoch_length,
-            false,
+            true,
```
Comment (Kouprin) [inline chain/client/tests/bug_repros.rs:160]: what is the reason of turning Doomslug on in the test?

Comment (bowenwang1996) [reply chain/client/tests/bug_repros.rs]: I was experimenting. It is not needed.

**Pull request merged**